### PR TITLE
Add pre-commit, format all files, and add justfile

### LINKED ISSUE #187: [Quality] Pre-commit hooks
## Problem
We have GitHub actions that expect a certain code style be adhered to, but this isn't enforced as a pre-commit hook, making it annoying for developers to have to fix these only after the action reports the problem.

## Description
Using the `pre-commit` package we can:

0. Add a pre-commit hook that will lint all our files
1. Replace the GitHub action script with a simple `pre-commit run --all-files`

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### LINKED ISSUE #193: [Quality] Use black and isort
## Current Situation
We do not use an automatic formatter.

## Suggested Improvement
Let's use an automatic formatter and import sorter so we don't have to think about it anymore. `black` is the standard opinionated Python formatter, and isort is the standard import sorter. Both are very safe and battle tested.

We will exclude the archive directory from them though.

## Benefit
We won't have to think about formatting any longer 🙂 

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #193 by @sarayourfriend 
Fixes #187 by @sarayourfriend 

## Description
Adds `pre-commit` using the same configuration as the `openverse-catalog` with some modifications to ignore rules in certain files that were problematic.

Also adds a `justfile` to make docker stuff easier.

## Technical details
I had to add some `noqa` comments for long strings (I wish flake8 would ignore strings...)

I also added `.flake8` that mirrors the pre-commit configuration for flake8 so that editor integration will work

## Tests
Checkout the branch and run `just install` then `just lint` and ensure that there are no errors reported.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    docker-compose exec web bash ./test/run_test.sh
+
+
+testlocal:
+    #! /usr/bin/env sh
+    cd openverse-api
+    pipenv run bash ./test/run_test.sh
+
+
+up:
```
Comment (AetherUnbound) [inline justfile:37]: Since these docker commands (or at least `up`) are dependencies for other recipes, do you mind moving these 3 to the top of the file?

**Review by AetherUnbound — APPROVED** [review]
Looks good! Just one thought for re-ordering

**Review by dhruvkb — APPROVED** [review]
This is so awesome! 👍

**Review by zackkrida — APPROVED** [review]

**Review by obulat — APPROVED** [review]
I missed 'just' when coming from the catalog repository :) Thank you for improving DX!

**Review by dhruvkb — COMMENTED** [review]
@sarayourfriend this PR does not update the `README.md` file in the root? #201 had that updated to match the `just` commands.

**Comment by sarayourfriend** [comment]
@dhruvkb whoops, I missed that when moving things over to this branch, I'll update the README now. Thanks for catching that!

**Comment by zackkrida** [comment]
such excellent work @sarayourfriend 🎉 can't wait to use these commands. thank you

**Pull request merged**